cmds/core/sluinit/uinit_linux.go

```diff
@@ -53,6 +100,21 @@ func checkDebugFlag() {
 	}
 }
 
+func parseBootEntryFlag() (string, error) {
+	bootEntry, present := cmdline.Flag(bootEntryFlag)
+	if !present {
+		return "", fmt.Errorf("no boot entry found")
+	}
+
+	if !launcher.IsValidBootEntry(bootEntry) {
+		return "", fmt.Errorf("boot entry contains bad characters: '%s'", bootEntry)
```

Comment:
here's where instead of '%s' you really want %q
	s := "\x03\x45ABD"
	fmt.Printf("'%s' %q\n", s, s)

'EABD' "\x03EABD"
is way more useful